### PR TITLE
Refactor components (Obsolete)

### PR DESCRIPTION
**Pull request by zackbraksa**
## Description
* Refactoring `BasicSide` component to use MUI Drawer components (List, ListItem, ListItemButton ...). 
* Refactoring `BasicHeader`, for now extracted `BasicAutocomplete` as a standalone React component.

### CODE REVIEW

File: src/lib/BasicAdmin.tsx [hunk 1]
```diff
@@ -3,16 +3,17 @@ import React from 'react'
...
 import { Gubu } from 'gubu'
 
-
-function BasicAdmin(props: any) {
+function BasicAdmin (props: any) {
   const {
     vxg,
     ctx,
-    spec,
+    spec
```
Comment (rjrodger) [inline src/lib/BasicAdmin.tsx:16]: leave trailing commas - they make it easier to do cut and paste reordering

File: src/lib/BasicAdmin.tsx [hunk 2]
```diff
@@ -30,33 +31,32 @@ function BasicAdmin(props: any) {
...
     foot: frameModel.part.foot,
-    view: frameModel.view,
+    view: frameModel.view
   }
 
   return (
     <div>
       <BasicHead vxg={vxg} ctx={ctx} spec={headSpec} />
-      <BasicSide vxg={vxg} ctx={ctx} spec={sideSpec} />
+      <BasicSidebar vxg={vxg} ctx={ctx} spec={sideSpec} />
```
Comment (rjrodger) [inline src/lib/BasicAdmin.tsx:55]: don't change the name - this needs to be consistent with the model

File: src/lib/BasicAppBar.tsx [hunk 3]
```diff
@@ -2,29 +2,27 @@ import MuiAppBar, { AppBarProps as MuiAppBarProps } from '@mui/material/AppBar'
 
 import { styled, useTheme, Theme, CSSObject } from '@mui/material/styles'
 
-
 interface AppBarProps extends MuiAppBarProps {
-  open?: boolean,
-  drawerwidth?: any,
+  open?: boolean
```
Comment (rjrodger) [inline src/lib/BasicAppBar.tsx:6]: in this case, no commas needed, and reordering is also easy

File: src/lib/BasicEdit.tsx [hunk 4]
```diff
@@ -73,121 +70,117 @@ function BasicEdit(props: any) {
...
+                            disabled={!field.edit}
+                            value={value}
+                            getOptionLabel={(option: any) => option || ''}
+                            filterOptions={(options: any, params: any) => {
+                              const filtered = filter(options, params)
+                              const { inputValue } = params
+                              // Suggest the creation of a new value
+                              const isExisting = options.some((option: any) => inputValue === option)
+
+                              if (inputValue != '' && !isExisting) {
```
Comment (rjrodger) [inline src/lib/BasicEdit.tsx:94]: prefer `'' !== inputValue`
see https://en.wikipedia.org/wiki/Yoda_conditions

File: src/lib/BasicHead.tsx [hunk 5]
```diff
@@ -4,23 +4,22 @@ import { useSelector } from 'react-redux'
...
+  createFilterOptions
+} from '@mui/material'
 
 import ChevronRightIcon from '@mui/icons-material/ChevronRight'
 
 import BasicButton from './BasicButton'
-
 import BasicAppBar from './BasicAppBar'
-
+import BasicAutocomplete from './Header/BasicAutoComplete'
```
Comment (rjrodger) [inline src/lib/BasicHead.tsx:22]: let's keep all the components in one folder for the moment

File: src/lib/BasicHead.tsx [hunk 6]
```diff
@@ -62,152 +49,119 @@ function BasicHead(props: any) {
...
   const {
-    frame,
+    frame
   } = spec
 
-  // console.log('BasicHead.spec', spec)
-
   // spec schema definition with Gubu
-  const shape = Gubu({
+  const shape = Gubu(Open({
```
Comment (rjrodger) [inline src/lib/BasicHead.tsx:56]: let's call this `BasicHeadSpecShape` and place outside the function, as we only need to create it once

File: src/lib/BasicMain.tsx [hunk 7]
```diff
@@ -8,33 +8,28 @@ import {
...
 
-
-function makeCmp(view: any, ctx: any) {
+function makeCmp (view: any, ctx: any) {
   let cmp: any = () => <div>NONE</div>
 
   const content = view.content || {}
 
-  if ('custom' === content.kind) {
+  if (content.kind === 'custom') {
```
Comment (rjrodger) [inline src/lib/BasicMain.tsx:19]: yoda

File: src/lib/BasicHead.tsx (hunk 6 above)
Comment (rjrodger) [reply src/lib/BasicHead.tsx:56]: and same for the other components

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: src/lib/BasicMain.tsx (hunk 7 above)
Comment (zackbraksa) [reply src/lib/BasicMain.tsx:19]: [`ts-standard`](https://github.com/standard/ts-standard) seems to be the one overwriting yoda conditions. And It doesn't seem it's possible to disable rules on `ts-standard` (according to their docs). 

Instead they recommend using `ESLint` with [`standard/eslint-config-standard-with-typescript`](https://github.com/standard/eslint-config-standard-with-typescript)

Should I switch to using `ESLint` with `standard/eslint-config-standard-with-typescript`? It should give us more flexibility while still leveraging the standard rules.

**Comment by zackbraksa** [comment]
Since the changes touch on multiple files, I will try to fix them in multiple smaller PRs. Hence why i'm closing this.

**Pull request closed without merging**